packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2145,7 +2145,7 @@ abstract class RenderBox extends RenderObject {
     assert(!_debugDoingBaseline, 'Please see the documentation for computeDistanceToActualBaseline for the required calling conventions of this method.');
     assert(!debugNeedsLayout);
     assert(() {
-      final RenderObject? parent = this.parent as RenderObject?;
+      final RenderObject? parent = this.parent;
```

Comment:
this might in fact be unnecessary now, the getter should just get inlined right?

Reply:
This is used in the asset on L2154 that asserts that the parent getter is stable